### PR TITLE
Add no_plot Boolean variable to gnomview

### PR DESCRIPTION
**Pull request by j-erler**
This commit adds the Boolean no_plot variable to the gnomview function in visufunc.py. Setting the variable to True will close the automatically generated output figure, which is useful is gnomview is run in a loop i.e. when stacking a large number of maps. Without this option, healpy will consume large amounts of RAM and can crash jupyter notebooks.

### CODE REVIEW

File: healpy/visufunc.py [hunk 1]
```diff
@@ -387,6 +388,8 @@ def gnomview(
       If True: do not add resolution info text. Default=False
     return_projected_map : bool
       if True returns the projected map in a 2d numpy array
+    no_plot : bool
```
Comment (DanielLenz) [inline healpy/visufunc.py]: Could you make it more consistent and use
`no_plot : bool, optional`?
(and please also add that to the `return_projected_map` keyword)

File: healpy/visufunc.py [hunk 2]
```diff
@@ -542,6 +545,10 @@ def gnomview(
         if wasinteractive:
             pylab.ion()
             # pylab.show()
+        if no_plot is True:
```
Comment (DanielLenz) [inline healpy/visufunc.py]: Just use `if no_plot:`

**Comment by DanielLenz** [comment]
Thanks for the PR! Are you aiming to generate many plots and save them without actually showing, or are you just interested in the return map you get from `return_projected_map`?

**Comment by j-erler** [comment]
Thanks for the suggestions on how to make the new code consistent with the previous. I was intending to use `no_plot` together with `return_projected_map` to create a data cube using a for-loop, after which the data could be saved. This requires the least modification to gnomview. Not sure if this feature would be useful to have with the other functions in `visufunc`, since those produce a single image from an all-sky map and I see no reason why you would want to run a loop or suppress the output.  I forgot to mention that this commit solves Issue #436 but is more of a hack than a proper solution since a figure is still created but immediately closed again. Preventing the creation of a figure in the first place would require much deeper modification.

**Comment by DanielLenz** [comment]
Thanks for adjusting that so quickly!

I'm not yet fully getting the other point though, sorry. When talking about a datacube, you mean you have several HEALPix maps at different frequencies/energies/colors, and you want to generate a cube for a small patch of the sky?

If so, why wouldn't that be useful for a different projection, e.g. `cartview`?

**Comment by DanielLenz** [comment]
Thanks for clarifying! We should wait for a few more opinions to see what the extent of this PR should be.

Either way, if you're only interested in reprojecting HEALPix maps to 2D maps, you should check out [`reproject`](https://reproject.readthedocs.io/en/stable/), which is really built for that.

**Comment by zonca** [comment]
thanks @j-erler and @DanielLenz

**Pull request merged**